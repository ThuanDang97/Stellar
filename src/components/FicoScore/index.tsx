import { FC, memo } from 'react'
import { useTheme } from 'styled-components'

// Components
import {
  BackDropPointLocker,
  BoxShadowStyled,
  TextStyled,
  ViewStyled,
} from '@components/styled-components'
import {
  EndBaseLineStyled,
  MiddleBaseLineStyled,
  StartBaseLineStyled,
  LineScoreStyled,
  ArrowDownStyled,
} from '@components/styled-components/LineStyled'
import { TextFadeInScoreStyled } from '@components/AnimatedText/AnimatedTextStyled'
import Ticker from '@components/Ticker'
import Title from '@components/Title'
import Link from '@components/Link'
import PlaceholderItem from '@components/PlaceholderItem'

// Props type
import { FicoScoreProps } from '@self-types/components/FicoScoreProps.props'

// Utils
import { handleCalFicoScore } from '@utils/index'
import { pxToRem } from '@utils/theme'

// Constants
import { OPACITY_BLUR, OPACITY_BOLD } from '@constants/index'

// Styles
import { AnimatedButtonUnlock } from '../../../styles/pages/DashBoardPageStyled'

const FicoScore: FC<FicoScoreProps> = ({
  isLocked,
  currentFicoScore,
  dateUpdated,
  minScore,
  maxScore,
}) => {
  const theme = useTheme()

  const locked = isLocked ? OPACITY_BLUR : OPACITY_BOLD

  const ficoScore = handleCalFicoScore(currentFicoScore)

  return (
    <BoxShadowStyled
      shadowType="normal"
      position="relative"
      width={theme.metrics.width.full}
      bgColor={theme.colors.white}
      minHeight={150}
      padding={`${
        isLocked
          ? `${pxToRem(theme.metrics.dimensions.xl)}`
          : `${pxToRem(theme.metrics.dimensions.md)}`
      } ${pxToRem(theme.metrics.dimensions.xlg)}`}
    >
      {isLocked && (
        <>
          {isLocked && (
            <AnimatedButtonUnlock
              position="absolute"
              mLeft={8}
              zIndex={theme.metrics.zIndex.low}
            >
              <Link
                href="/free-credit-score"
                text="Unlock Credit Score"
                linkTypes="unLockLink"
                iconLeftUrl="/icons/unlock.svg"
                width={230}
              />
            </AnimatedButtonUnlock>
          )}
          <BackDropPointLocker
            isLocked
            display="flex"
            alignItems="flex-start"
            width={260}
            position="relative"
            zIndex={theme.metrics.zIndex.under}
          >
            <TextStyled
              fontSize={theme.typography.fontSize.xxl}
              lineHeight={theme.typography.lineHeight.md}
              color={theme.colors.mercury}
            >
              728
            </TextStyled>
            <ViewStyled mLeft={theme.metrics.dimensions.lg}>
              <PlaceholderItem width={90} height={15} />
              <PlaceholderItem
                width={theme.metrics.width.lg}
                height={13}
                mTop={theme.metrics.dimensions.sm}
              />
            </ViewStyled>
          </BackDropPointLocker>
        </>
      )}
      {!isLocked && (
        <ViewStyled display="flex" position="relative">
          <Ticker
            size={53}
            firstNumber={currentFicoScore}
            duration={1}
            delay={0.1}
          />
          <ViewStyled
            display="flex"
            flexDirection="column"
            alignItems="initial"
            mLeft={theme.metrics.dimensions.base}
          >
            <ViewStyled display="flex">
              <Title
                headingLevel="h2"
                fontWeight={theme.typography.fontWeight.bold}
                color={theme.colors.black}
                fontSize={theme.typography.fontSize.lg}
                lineHeight={theme.typography.lineHeight.md}
                textAlign="left"
                title="FICO"
              />
              <Title
                headingLevel="h2"
                color={theme.colors.black}
                fontSize={theme.typography.fontSize.lg}
                fontFamily="AdobeCleanLight"
                title=" SCORE 8"
                fontWeight={theme.typography.fontWeight.base}
              />
            </ViewStyled>

            <TextStyled
              color={theme.colors.cello}
              fontSize={theme.typography.fontSize.xss}
              fontFamily="AdobeCleanLight"
              textAlign="left"
              lineHeight={theme.metrics.dimensions.tiny}
              letterSpacing={theme.typography.letterSpacing.xs}
            >
              {dateUpdated}
            </TextStyled>
          </ViewStyled>
        </ViewStyled>
      )}

      <ViewStyled
        display="flex"
        mTop={theme.metrics.dimensions.xl}
        position="relative"
      >
        {!isLocked && (
          <TextStyled
            as="p"
            color={theme.colors.black}
            fontSize={theme.typography.fontSize.base}
            position="absolute"
            left={-25}
            mTop={-4}
          >
            {minScore}
          </TextStyled>
        )}
        <StartBaseLineStyled
          bgColor={theme.colors.white}
          lineColor={theme.colors.pomegranate}
          width={51}
          height={7}
          border={3}
          mLeft={4}
          opacity={locked}
        />
        <MiddleBaseLineStyled
          bgColor={theme.colors.white}
          width={16}
          lineColor={theme.colors.pizzaz}
          opacity={locked}
          height={7}
          border={3}
        />
        <MiddleBaseLineStyled
          bgColor={theme.colors.white}
          width={14}
          lineColor={theme.colors.supernova}
          position="relative"
          opacity={locked}
          height={7}
          border={3}
        />
        <EndBaseLineStyled
          bgColor={theme.colors.white}
          width={20}
          lineColor={theme.colors.caribbeanGreen}
          position="relative"
          opacity={locked}
          height={7}
          border={3}
        />

        <>
          <LineScoreStyled
            width={isLocked ? 14 : 100 - ficoScore.percentLineScore}
            opacity={0.7}
          />
          {!isLocked && (
            <>
              <TextStyled
                as="p"
                color={theme.colors.black}
                fontSize={theme.typography.fontSize.base}
                position="absolute"
                left={theme.metrics.width.full}
                mLeft={4}
                mTop={-4}
              >
                {maxScore}
              </TextStyled>
              <ArrowDownStyled
                left={ficoScore.positionFicoScore}
                lineColor={ficoScore.color}
                delay={1}
              >
                <TextFadeInScoreStyled
                  color={ficoScore.color}
                  fontSize={theme.typography.fontSize.xs}
                  mBottom={theme.metrics.dimensions.xs}
                  duration={0.3}
                >
                  {ficoScore.text}
                </TextFadeInScoreStyled>
              </ArrowDownStyled>
            </>
          )}
        </>
      </ViewStyled>
    </BoxShadowStyled>
  )
}

export default memo(FicoScore)
