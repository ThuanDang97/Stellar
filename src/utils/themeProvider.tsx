import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
