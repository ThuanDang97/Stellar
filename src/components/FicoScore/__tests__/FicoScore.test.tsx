import React from 'react'

// Components
import FicoScore from '@components/FicoScore'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('FicoScore component', () => {
  it('renders FicoScore correctly', () => {
    const { asFragment } = renderWithTheme(
      <FicoScore
        isLocked={false}
        currentFicoScore={0}
        dateUpdated=""
        minScore={300}
        maxScore={850}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
