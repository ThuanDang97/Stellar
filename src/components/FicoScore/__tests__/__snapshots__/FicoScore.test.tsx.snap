// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FicoScore component renders FicoScore correctly 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 kVCtEV btvkUa"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 iGWxQB"
      display="flex"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 TickerStyled__TickerWrapperStyled-sc-x8b1pu-0 eQdhWe inONBg"
        color="#F22E3E"
        data-testid="digit-container"
        height="0"
      >
        <div
          class="TickerStyled__TickerColumn-sc-x8b1pu-1 TickerStyled__TickerColumnOne-sc-x8b1pu-2 fPgOvX duqavd"
        >
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
        </div>
        <div
          class="TickerStyled__TickerColumn-sc-x8b1pu-1 TickerStyled__TickerColumnTwo-sc-x8b1pu-3 eYwLsI JKGqo"
        >
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
        </div>
        <div
          class="TickerStyled__TickerColumn-sc-x8b1pu-1 TickerStyled__TickerColumnThree-sc-x8b1pu-4 eYwLsI lpsVtF"
        >
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            0
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            1
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            2
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            3
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            4
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            5
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            6
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            7
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            8
          </h2>
          <h2
            class="TickerStyled__TickerDigit-sc-x8b1pu-5 iimuNt"
            font-size="53"
          >
            9
          </h2>
        </div>
      </div>
      <div
        class="TickerStyled__TickerSimulator-sc-x8b1pu-6 cSXHfs"
        font-size="53"
      >
        0
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 jhgBWe"
        display="flex"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 knVWgS"
          display="flex"
        >
          <h2
            class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 dozpQ fTabgG"
            color="#000"
            font-size="24"
            font-weight="700"
          >
            FICO
          </h2>
          <h2
            class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 bZddvp eABfQQ"
            color="#000"
            font-family="AdobeCleanLight"
            font-size="24"
            font-weight="400"
          >
             SCORE 8
          </h2>
        </div>
        <span
          class="TextStyled-sc-9xia41-0 leGgIj"
          color="#203F59"
          font-family="AdobeCleanLight"
          font-size="13"
          letter-spacing="0.4"
        />
      </div>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 cEttAk"
      display="flex"
    >
      <p
        class="TextStyled-sc-9xia41-0 iqtaRc"
        color="#000"
        font-size="14"
      >
        300
      </p>
      <div
        class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__StartBaseLineStyled-sc-okjczv-1 eiaXsV ifLECW"
        height="7"
        opacity="1"
        width="51"
      />
      <div
        class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__MiddleBaseLineStyled-sc-okjczv-2 duaSFb dbuCnc"
        height="7"
        opacity="1"
        width="16"
      />
      <div
        class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__MiddleBaseLineStyled-sc-okjczv-2 fssdGi dbIHPc"
        height="7"
        opacity="1"
        width="14"
      />
      <div
        class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__EndBaseLineStyled-sc-okjczv-3 kIORHB fKiEZo"
        height="7"
        opacity="1"
        width="20"
      />
      <div
        class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__LineScoreStyled-sc-okjczv-4 geSzQn hjZOmd"
        opacity="0.7"
        width="153"
      />
      <p
        class="TextStyled-sc-9xia41-0 ljPOft"
        color="#000"
        font-size="14"
      >
        850
      </p>
      <div
        class="LineStyled__ArrowDownStyled-sc-okjczv-5 bONNmq"
      >
        <span
          class="TextStyled-sc-9xia41-0 AnimatedTextStyled__TextFadeInScoreStyled-sc-1e3pr1k-5 hVOfzN RJerm"
          color="#00D89D"
          font-size="11"
        >
          Excellent
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
